lib/offline.js

```diff
@@ -119,19 +131,30 @@ module.exports = function(config) {
         var data = fs.readFileSync(filepath);
         totalGzipSize += gzipSize.sync(data);
         var hash = getHash(data);
-        process.stdout.write(chalk.green.bold('✓ ') + 'Caching ' + filepath + ' (' + prettyBytes(size) + ')\n');
+        var path = getPathName(filepath, rootDir);
+        var rootMatch = new RegExp('^' + rootDir);
+        var pathAtString = '';
+        if (filepath.replace(rootMatch, './') !== path) {
+          pathAtString = ' at path ' + path;
+        }
+
+        process.stdout.write(chalk.green.bold('✓ ') + 'Caching ' + filepath + pathAtString + ' (' + prettyBytes(size) + ')\n');
         return {
-          path: filepath.replace(rootDir, function (match, offset) {
-            // The root must be the worker's directory
-            return offset === 0 ? './' : match;
-          }),
+          path: path,
           hash: hash,
         };
       });
 
       process.stdout.write('Total cache size is ' + prettyBytes(totalSize) + ' (' + prettyBytes(totalGzipSize) + ' if served with gzip) for ' + files.length + ' files.\n');
 
-      return filesAndHashes;
+      filesAndHashes.unshift({
+        path: './'
+      }); // cache always the current root to make the default page available
+
+      // Remove duplicates from file list
+      return filesAndHashes.filter(function(fileName, index) {
+        return filesAndHashes.indexOf(fileName) === index;
+      });
```

Comment:
This certainly makes sense in a setup where you can specify to prefer the `index.html` over the `/` directory path. However I will await your response on my latest comment which should hopefully clear up the confusion.